datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -631,6 +638,12 @@ pub mod proto {
         #[prost(uint64, tag = "1")]
         pub k: u64,
     }
+
+    #[derive(Clone, PartialEq, ::prost::Message)]
+    pub struct ScalarUDFProto {
+        #[prost(message, tag = "1")]
+        pub expr: ::core::option::Option<datafusion_proto::protobuf::LogicalExprNode>,
+    }
```

Comment:
Here is my test proto

Reply:
This is not how this would work. Expressions would be passed to the udf as arguments. What would potentially get serialized into `ScalarUDFProto` would just be some non-expr state. For example, a regex UDF might look like:

```
struct MyRegexUdf {
  // compiled regex
  regex: Regex,
  // regex as original string
  pattern: String,
}

impl ScalarUdfImpl for MyRegexUdf {
 // implementation here, not important for example
}
```

This would take a `Utf8` expression in as an argument when evaluated but that will be serialized and deserialized separately. In the proto we only need to serialized the regex. So the proto would be something like

```
message MyRegexUdfNode {
  string pattern = 1;
}
```

Then in our custom codec we have

```
fn try_decode_udf(&self, name: &str, buf: &[u8]) -> Result<Arc<ScalarUDF>> {
    if let Ok(proto) = MyRegexUdfNode::decode(buf) {
      let regex = Regex::new(&proto.pattern).map_err(|err| ...)?;
      Ok(Arc::new(
        ScalarUDF::new_from_impl(MyRegexUdf {
           regex,
           pattern: proto.pattern,
         })
      ))
    } else {
      not_impl_err!("unrecognized scalar UDF implementation, cannot decode")
    }
}
```